Std/Tactic/Ext.lean

```diff
@@ -98,6 +104,8 @@ def applyExtLemma (goal : MVarId) : MetaM (List MVarId) := goal.withContext do
         let c ← mkConstWithFreshMVarLevels lem.declName
         let (_, _, declTy) ← withDefault <| forallMetaTelescopeReducing (← inferType c)
         guard (← isDefEq tgt declTy)
+      if tactic.ext.trace.get (← getOptions) then
+        logInfo s!"try: apply {lem.declName}"
```

Comment:
Is there a convenient way to bundle multiple log messages into one? Would need to be in `withExtN` and I don't know how to do that besides just adding the log string as a further argument to all the functions, which seems inconvenient.

Maybe using `Std.Tactic.TryThis` somehow?

Reply:
This depends on what the interface you are going for is, but yes I would expect this to involve passing the accumulated trace through `withExtN`. You can also do things with IO.Ref if you want to pass state through a combinator that doesn't support it but runs in a monad supporting IO.